# 1) Importation Tweet traitant du Bitcoin :

In [28]:
import twitter

In [3]:
api = twitter.Api(consumer_key='WfQELhZV1qiIvYVfBc5wqO5xr',
                      consumer_secret='wdldaj2Hw6h7nRp2KTYYOLk7kkmkPA6SfXdwAppYoIQa5Z19P1',
                      access_token_key='1174300508384911365-U8PxqaeUtfyy7jV4Rwtc9DTU2IQj7L',
                      access_token_secret='zIoV392S1t062tFprvGhWwkqtCla5iMSRJdzYZa8VO7at')

In [10]:
previous_Id = 1000041612492296192
result_Text = []
for i in range(1,149) :
    next_Id= previous_Id + 400000000000000000
    results = api.GetSearch(
        raw_query=f"q=Bitcoin&result_type=mixed&since=2019-09-02&count=10000&lang=en&tweet_mode=extended&max_id={next_Id}&since_id={previous_Id}")
    for j in results :
        if(j.retweet_count>2):
            result_Text.append(j.full_text)
    previous_Id = next_Id


# 2) Sauvegarde des tweets dans un .txt

In [29]:
file = open('08_10_19.txt',"w", encoding='utf-8')
for text in result_Text :
    print(text)
    text = text
    file.write(text)
file.close()

RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @Rhythmtrader: On average, $2,000,000,000 worth of value is transfered every single day using the bitcoin network.

No government, bank…
RT @Rhythmtrader: On average, $2,000,000,000 worth of value is transfered every single day using the bitcoin network.

No government, bank…
RT @rogerkver: Bitcoin Cash is the second most used crypto currency for payments in the world.
https://t.co/HccxMNbK7P
RT @dergigi: "We are all bitcoin" -- @HassMcCook
RT @Bitcoin: Recently:

1. Turkey froze 3m+ bank accounts.
2. Hong Kong ATM withdrawals restricted.
3. Reserve Bank of India ordered PMC Ba…
RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @real_vijay: Keynesian econstrology has become so pervasive that it's widely believed spending is more important to saving for

RT @Bitcoin: Recently:

1. Turkey froze 3m+ bank accounts.
2. Hong Kong ATM withdrawals restricted.
3. Reserve Bank of India ordered PMC Ba…
RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @real_vijay: Keynesian econstrology has become so pervasive that it's widely believed spending is more important to saving for the healt…
RT @CryptoJungle: Origins and development of $VEO, @zack_bitcoin's unique perspective, PoW vs PoS, oracles.

The 4th installment of The Und…
RT @CoinGuru247: To win 20,000 BTT coins you MUST:

✅Like &amp; Retweet
✅Follow @CoinGuru247
✅Tag 3 Friends

1 random winner, 
Giveaway ends Oc…
RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @newsbtc: Bitcoin Price Analysis by @AayushJs: Bitcoin price is currently rebounding above $8,200 against the US Dollar. However, $BTC s…
RT @xrpre

RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @real_vijay: Keynesian econstrology has become so pervasive that it's widely believed spending is more important to saving for the healt…
RT @CryptoJungle: Origins and development of $VEO, @zack_bitcoin's unique perspective, PoW vs PoS, oracles.

The 4th installment of The Und…
RT @CoinGuru247: To win 20,000 BTT coins you MUST:

✅Like &amp; Retweet
✅Follow @CoinGuru247
✅Tag 3 Friends

1 random winner, 
Giveaway ends Oc…
RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
RT @newsbtc: Bitcoin Price Analysis by @AayushJs: Bitcoin price is currently rebounding above $8,200 against the US Dollar. However, $BTC s…
RT @xrpreality: We are witnessing the new era, people will start selling all other alts to buy XRP because the value of the token will rise…
RT @Crypt

RT @bitangler: 🥳 ($70,000) #BITCOIN GIVEAWAY 🥳

I’m up almost 80k this week from holding BTC, so I decided to give some away to my supporte…
[New Post] Bitcoin is becoming much more important in Hong Kong and India. 

It won't happen overnight, but slowly people are reminded why they should own a decentralized digital currency.

https://t.co/lJHMyeeJKe
Bitcoin is different things to different people.

When you’re worried about your assets being seized or becoming inaccessible to you, Bitcoin’s non-seizability becomes very attractive.

This aspect of Bitcoin just became important for 1+ billion people in India &amp; Hong Kong. https://t.co/nLLaUlRpBz
General Electric has frozen pension payments for 20,000 employees.

The pension problem in America is only going to get worse over time.

Promises will be broken.

Every pension needs to have exposure to Bitcoin.

They must #GetOffZero 🙏🏽
RT @markethive: Twitter isn’t tired of Bitcoin even though hashtags are at an all-time ...
Found on:
Ma

# 3) Importation des packages pour la première version de NLP :

In [30]:
import nltk
from nltk.tag import UnigramTagger
from nltk.corpus import treebank
train_sents = treebank.tagged_sents()[:3000]
tagger = UnigramTagger(train_sents)
from nltk.corpus import sentiwordnet as swn

# 4) Création Fonction première version NLP
## - Création Fonction pour étudier les émotions seulement sur les adverbes et adjectifs

Liste des tags possibles pour chaque mots : http://courses.washington.edu/hypertxt/csar-v02/penntable.html

In [52]:
def EtudierEmotionTexteOnlyAdvAndAdj(url) :
    fichierTexte = open(f"{url}",encoding='utf-8' )
    text = fichierTexte.read()
    
# Suppression de la ponctuation
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    text = text.replace('@','')
    
    text = text.split(' ')
    
# Suppression tous les types de mots excepté les adverbes et les adjectifs
    result_type_text = tagger.tag(text)
    liste_mots_a_conserver = []
    for i in range(len(result_type_text)) :
        if result_type_text[i][1] in ["JJ", "JJR", "JJS", "RBR", "RBS"] :
            liste_mots_a_conserver.append(result_type_text[i][0])
    
# Récupération d'une liste d'émotion pour chaque mots
    sentiments_par_mots = []
    for i in liste_mots_a_conserver :
        sentiments_par_mots.append(swn.senti_synsets(i))
    
# Création d'une liste avec l'émotion principale de chaque mots
    keepFirstOne = []
    for i in sentiments_par_mots :
        keepInMemory = i
        listI = list(i)
        if(len(listI) != 0) :
            keepFirstOne.append(listI[0])
        i = keepInMemory
    
# Somme des mots positifs et négatifs
    sommePos = 0
    sommeNeg = 0
    for i in keepFirstOne :
        if(i.pos_score()>i.neg_score()) :
            sommePos = sommePos + i.pos_score()
        sommeNeg = sommeNeg + i.neg_score()
    print(f"{sommePos} {sommeNeg}")
    
# Retourne le résultat
    if sommePos > sommeNeg : 
        return "Pos"
    else :
        return "Neg"

In [53]:
def EtudierEmotionTexteOnlyAdvAndAdjAllSenti(url) :
    fichierTexte = open(f"{url}",encoding='utf-8' )
    text = fichierTexte.read()
    
# Suppression de la ponctuation
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    text = text.replace('@','')
    
    text = text.split(' ')
    
# Suppression tous les types de mots excepté les adverbes et les adjectifs
    result_type_text = tagger.tag(text)
    liste_mots_a_conserver = []
    for i in range(len(result_type_text)) :
        if result_type_text[i][1] in ["JJ", "JJR", "JJS", "RBR", "RBS"] :
            liste_mots_a_conserver.append(result_type_text[i][0])
    
# Récupération d'une liste d'émotion pour chaque mots
    sentiments_par_mots = []
    for i in liste_mots_a_conserver :
        sentiments_par_mots.append(swn.senti_synsets(i))
    
# Création d'une liste avec l'émotion principale de chaque mots
#    keepFirstOne = []
#    for i in sentiments_par_mots :
#        keepInMemory = i
#        listI = list(i)
#        if(len(listI) != 0) :
#            keepFirstOne.append(listI[0])
#        i = keepInMemory
    
# Somme des mots positifs et négatifs
    sommePos = 0
    sommeNeg = 0
    for sentiments in sentiments_par_mots :
        for i in list(sentiments) :
            if(i.pos_score()>i.neg_score()) :
                sommePos = sommePos + i.pos_score()
            sommeNeg = sommeNeg + i.neg_score()
    print(f"{sommePos} {sommeNeg}")
    
# Retourne le résultat
    if sommePos > sommeNeg : 
        return "Pos"
    else :
        return "Neg"

## - Fonction pour étudier les émotions sur la totalité du texte

In [36]:
def EtudierEmotionTexte(url) :
    tes = open(f"{url}",encoding='utf-8' )
    text = tes.read()
    
# Suppression de la ponctuation
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    text = text.split(' ')
    
# Utile si on veut supprimer certains types de mots 
    result_type_text = tagger.tag(text)
    liste_mots_a_conserver = []
    for i in range(len(result_type_text)) :
            liste_mots_a_conserver.append(result_type_text[i][0])
    
# Récupération d'une liste d'émotion pour chaque mots
    sentiments_par_mots = []
    for i in liste_mots_a_conserver :
        sentiments_par_mots.append(swn.senti_synsets(i))
    
# Création d'une liste avec l'émotion principale de chaque mots
    keepFirstOne = []
    for i in sentiments_par_mots :
        keepInMemory = i
        listI = list(i)
        if(len(listI) != 0) :
            keepFirstOne.append(listI[0])
        i = keepInMemory
    
# Somme des mots positifs et négatifs
    sommePos = 0
    sommeNeg = 0
    for i in keepFirstOne :
        if(i.pos_score()>i.neg_score()) :
            sommePos = sommePos + i.pos_score()
        sommeNeg = sommeNeg + i.neg_score()
    print(f"{sommePos} {sommeNeg}")
    
# Retourne le résultat
    if sommePos > sommeNeg : 
        return "Pos"
    else :
        return "Neg"

# 5) Execution des méthodes et résultats :

In [54]:
print(EtudierEmotionTexteOnlyAdvAndAdj("08_10_19.txt"))

647.25 204.375
Pos


In [55]:
print(EtudierEmotionTexteOnlyAdvAndAdjAllSenti("08_10_19.txt"))

5836.027999999997 4009.8470000000034
Pos


In [21]:
print(EtudierEmotionTexte("08_10_19.txt"))

2607.75 1226.375
Pos


## - Résultats :

## Le 27/09/2019
#### 335.125 120.75 seulement adverbe et adjectif
#### 3422.75 1677.625 tout le texte
#### 3684.125 1879.375 Suppression de la ponctuation

---------------------------------------------------------------

## Le 30/09/2019 
#### 3071.25 2482.081999999997 Suppression de la ponctuation
#### 458 355.5 Tout le texte

----------------------------------------------------------------------

## Le 07/10/2019
#### 6157.25 4552.375 Suppression de la ponctuation

# 6) Version test du package VaderSentiment

In [22]:
from vaderSentiment.vaderSentiment import SentimentIntensityAnalyzer
analyser = SentimentIntensityAnalyzer()
import pandas as pd

In [33]:
def EtudierEmotionTexteR(url) :
    sentiment = []
    tes = open(f"{url}",encoding='utf-8' )
    text = tes.read()
    # Suppression de la ponctuation
    text = text.replace('.','')
    text = text.replace('!','')
    text = text.replace(',','')
    text = text.replace('?','')
    text = text.replace('#','')
    
    text = text.split(' ')
    # Récupérer les émotions mots par mots
    for tweet in text :
        score = analyser.polarity_scores(tweet)
        sentiment.append(score)
        
    tweetScore = pd.DataFrame(sentiment)
    print(tweetScore.pos.sum())
    print(tweetScore.neg.sum())
    print(tweetScore.mean())

In [27]:
EtudierEmotionTexteR("08_10_19.txt")

6720.516999999999
2295.1400000000003
compound    0.019061
neg         0.021085
neu         0.883425
pos         0.061739
dtype: float64


## Le 27/09 :
###### Number Pos : 5930.544999999998
###### Number Neg : 1695.179
###### compound    0.013088
###### neg         0.012271
###### neu         0.856211
###### pos         0.042930

---------------------------------------------------------

## Le 30/09 :
##### 8432.651000000002
##### 2028.686
##### compound    0.017799
##### neg         0.013978
##### neu         0.873808
##### pos         0.058104

-----------------------------------------------------------

## Le 02/10
##### Number Pos : 9964.012999999997
##### Number Neg : 3340.5380000000005
##### compound    0.010518
##### neg         0.014445
##### neu         0.883240
##### pos         0.043086

-----------------------------------------------------

## Le 07/10 
##### Number Pos: 10241.733000000004
##### Number Neg : 5652.090000000001
##### compound    0.007918
##### neg         0.022810
##### neu         0.871410
##### pos         0.041333

----------------------------------------------------------------------

## Le 08/10
##### Number Pos : 6720.516999999999
##### Number Neg : 2295.1400000000003
##### compound    0.019061
##### neg         0.021085
##### neu         0.883425
##### pos         0.061739